provider/pkg/provider/provider_plugin_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"context"
+	_ "embed"
+
+	pbempty "github.com/golang/protobuf/ptypes/empty"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+)
+
+var _ = Describe("RPC:Cancel", func() {
+	var k *kubeProvider
+
+	JustBeforeEach(func() {
```

Comment:
To get some clarity on Ginkgo lifecycle, would using BeforeEach here also yield the same effect?

Reply:
Yes, I'm using the same pattern of deferring `NewProvider` to be able to customize the opts.